docs/src/getting_started/teapot_rendering.md

```diff
@@ -0,0 +1,179 @@
+# Introduction to rendering using RayTracer.jl
+
+In this example we will render the famous UTAH Teapot model.
+We will go through the entire rendering API. We will load
+an obj file for the scene. This needs to be downloaded manually.
+
+Run this code in your terminal to get the file:
+`wget https://raw.githubusercontent.com/McNopper/OpenGL/master/Binaries/teapot.obj`
+
+If you are using REPL mode you need the `ImageView.jl` package
+
+```julia
+using RayTracer, Images #, ImageView
+```
+
+## General Attributes of the Scene
+
+Specify the dimensions of the image we want to generate.
+`screen_size` is never passed into the RayTracer directly so it
+need not be a named tuple.
+
+```julia
+screen_size = (w = 256, h = 256)
+```
+
+Load the teapot object from an `obj` file. We can also specify
+the scene using primitive objects directly but that becomes a
+bit involved when there are complicated objects in the scene.
+
+```julia
+scene = load_obj("teapot.obj")
+```
+
+We shall define a convenience function for rendering and saving
+the images.
+For understanding the parameters passed to the individual functions
+look into the documentations of [`get_primary_rays`](@ref), [`raytrace`](@ref)
+and [`get_image`](@ref)
+
+```julia
+function generate_render_and_save(cam, light, filename)
+    #src # Get the primary rays for the camera
```

Comment:
What are these `#src`?

Reply:
These prevent literate from splitting the function block into multiple code blocks